src/V3AssertPre.cpp

```diff
@@ -139,16 +152,195 @@ class AssertPreVisitor final : public VNVisitor {
     // VISITORS
     //========== Statements
     void visit(AstClocking* nodep) override {
+        m_clockingp = nodep;
         UINFO(8, "   CLOCKING" << nodep << endl);
-        // Store the new default clock, reset on new module
-        m_seniDefaultp = nodep->sensesp();
-        // Trash it, keeping children
-        if (nodep->bodysp()) {
-            nodep->replaceWith(nodep->bodysp()->unlinkFrBack());
+        iterateChildren(nodep);
+    }
+    void visit(AstClockingItem* nodep) override {
+        auto* const flp = nodep->fileline();
+        V3Const::constifyEdit(nodep->skewp());
+        if (!VN_IS(nodep->skewp(), Const)) {
+            nodep->skewp()->v3error("Skew must be constant (IEEE 1800-2017 14.4)");
+            VL_DO_DANGLING(nodep->unlinkFrBack()->deleteTree(), nodep);
+            return;
+        }
+        auto* const skewp = VN_AS(nodep->skewp(), Const);
+        if (skewp->num().isNegative()) skewp->v3error("Skew cannot be negative");
+        auto* const exprp = nodep->exprsp();
+        // Get a ref to the sampled/driven variable
+        auto* const varp = nodep->varp();
+        varp->unlinkFrBack();
+        m_clockingp->addVarsp(varp);
+        varp->user1p(nodep);
+        if (nodep->direction() == VDirection::OUTPUT) {
+            auto* const skewedRefp = new AstVarRef{flp, varp, VAccess::READ};
+            skewedRefp->user1(true);
+            auto* const assignp = new AstAssign{flp, exprp->cloneTree(false), skewedRefp};
+            if (skewp->isZero()) {
+                // Drive the var in Re-NBA (IEEE 1800-2017 14.16)
+                m_clockingp->addNextHere(new AstAlwaysReactive{
+                    flp, new AstSenTree{flp, m_clockingp->sensesp()->cloneTree(false)}, assignp});
+            } else if (skewp->fileline()->timingOn()) {
+                auto* const forkp = new AstFork{flp, "", assignp};
```

Comment:
Do we really need fork in every likely usage? That's very expensive.

Reply:
Without a fork, this always couldn't be retriggered before the assignment happens. Also this way it can be scheduled as combo, otherwise a new trigger would have to be created.

I'm not sure performance is a huge concern, how many signals are usually driven via clocking blocks?

Reply:
I was assuming that most clocking blocks that activate on an edge can be transformed into normal statements not requiring fork?

Reply:
They activate on edge, but if a clockvar has output skew then that is represented as an assignment with a delay equal to the skew. Leading to my previous comment.

Reply:
Ok, I'm with you now, thanks.